src/explore-education-statistics-admin/src/pages/release/data/components/AncillaryFileForm.tsx

```diff
@@ -0,0 +1,171 @@
+
+export interface AncillaryFileFormValues {
+  title: string;
+  summary: string;
+  file: File | null;
+}
+
+const formId = 'ancillaryFileForm';
+
+const errorMappings = [
+  mapFieldErrors<AncillaryFileFormValues>({
+    target: 'title',
+    messages: {
+      FileUploadNameCannotContainSpecialCharacters:
+        'File upload name cannot contain special characters',
+    },
+  }),
+  mapFieldErrors<AncillaryFileFormValues>({
+    target: 'file',
+    messages: {
+      CannotOverwriteFile: 'Choose a unique file name',
+      FileCannotBeEmpty: 'Choose a file that is not empty',
+      FileTypeInvalid: 'Choose a file of an allowed format',
+      FilenameCannotContainSpacesOrSpecialCharacters:
+        'Filename cannot contain spaces or special characters',
+    },
+  }),
+];
+
+export interface AncillaryFileFormProps {
+  files?: AncillaryFile[];
+  fileFieldLabel?: string;
+  initialValues?: AncillaryFileFormValues;
+  resetAfterSubmit?: boolean;
+  submitText?: string;
+  submittingText?: string;
+  validationSchema?: ObjectSchema<Partial<AncillaryFileFormValues>>;
+  onCancel?: () => void;
+  onSubmit: (values: AncillaryFileFormValues) => void;
+}
+
+export default function AncillaryFileForm({
+  files = [],
+  fileFieldLabel = 'Upload file',
+  initialValues,
+  resetAfterSubmit,
+  submitText = 'Save file',
+  submittingText = 'Saving file',
+  validationSchema,
+  onCancel,
+  onSubmit,
+}: AncillaryFileFormProps) {
+  const handleSubmit = useFormSubmit<AncillaryFileFormValues>(
+    async (values, actions) => {
+      await onSubmit(values);
+
+      if (resetAfterSubmit) {
+        actions.resetForm();
+      }
+    },
+    errorMappings,
+  );
+
+  return (
+    <Formik<AncillaryFileFormValues>
+      enableReinitialize
+      initialValues={
+        initialValues ?? {
+          title: '',
+          summary: '',
+          file: null,
+        }
+      }
+      onReset={() => {
+        document
+          .querySelectorAll(`#${formId} input[type='file']`)
+          .forEach(input => {
+            const fileInput = input as HTMLInputElement;
+            fileInput.value = '';
+          });
+      }}
+      onSubmit={handleSubmit}
+      validationSchema={Yup.object<AncillaryFileFormValues>({
+        title: Yup.string()
+          .trim()
+          .required('Enter a title')
+          .test({
+            name: 'unique',
+            message: 'Enter a unique title',
+            test(value: string) {
+              if (!value) {
+                return true;
+              }
+
+              return files.every(
+                f => f.title.toUpperCase() !== value.toUpperCase(),
+              );
+            },
+          }),
+        summary: Yup.string().required('Enter a summary'),
+        file: Yup.file().minSize(0, 'Choose a file that is not empty'),
+      }).concat(validationSchema ?? Yup.object())}
+    >
+      {form => (
+        <Form id={formId}>
+          <div style={{ position: 'relative' }}>
```

Comment:
Is this div needed? I can't see any difference if I remove it.

Reply:
Removed now. This was just a hangover from previously using an overlay loading spinner on submit.